File: source/windowsMobile/CommLayer/SubProtocolList.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Collections;

namespace CommLayer
{
    /// <summary>
    /// Lista de SubProtocolos
    /// </summary>
    public class SubProtocolList
    {
        /// <summary>
        /// La tabla de hashing de la colección
        /// </summary>
        internal Hashtable Collection;

        /// <summary>
        /// Candado para control de threading
        /// </summary>
        private Object thisLock;

        /// <summary>
        /// Default Constructor
        /// </summary>
        public SubProtocolList()
        {
            Collection = new Hashtable();
            thisLock = new Object();
        }

        /// <summary>
        /// Agrega un sub protocolo a la lista
        /// </summary>
        /// <param name="subProtocolType">El tipo del subprotocolo (los numeros entre 0 y 1000 esta reservados para el sistema)</param>
        /// <param name="subProtocol">un objeto de sub protocolo inicializado</param>
        public void add(Int32 subProtocolType, SubProtocolI subProtocol) 
        {
            lock (thisLock)
            {
                //Si no tengo el ip, entonces agrego al usuario como alguien nuevo
                if (!Collection.Contains(subProtocolType))
                {
                    Collection.Add(subProtocolType, subProtocol);
                }
                //Si ya la tengo, actualizo el objeto usuario
                else
                {
                    Collection.Remove(subProtocolType);
                    Collection.Add(subProtocolType, subProtocol);
                }
            }
        }
    }
}

File: source/java/hlmp/CommLayer/SubProtocolList.java
package hlmp.CommLayer;

import java.util.HashMap;

public class SubProtocolList {
	
	/**
     * La tabla de hashing de la colección
     */
    private HashMap<Integer, SubProtocol> Collection;


    /**
     * Default Constructor
     */
    public SubProtocolList()
    {
        Collection = new HashMap<Integer, SubProtocol>();
    }
    
    /**
     * Agrega un sub protocolo a la lista
     * @param subProtocolType El tipo del subprotocolo (los numeros entre 0 y 1000 esta reservados para el sistema)
     * @param subProtocol un objeto de sub protocolo inicializado
     */
    public synchronized void add(int subProtocolType, SubProtocol subProtocol) 
    {
            //Si no tengo el ip, entonces agrego al usuario como alguien nuevo
            if (!Collection.containsKey(subProtocolType))
            {
                Collection.put(subProtocolType, subProtocol);
            }
            //Si ya la tengo, actualizo el objeto usuario
            else
            {
                Collection.remove(subProtocolType);
                Collection.put(subProtocolType, subProtocol);
            }
    }

	public HashMap<Integer, SubProtocol> getCollection() {
		return Collection;
	}

}
